File: WPF Databinding in Visual Studio 2010 RTM/[C#]-WPF Databinding in Visual Studio 2010 RTM/C#/WpfEfSampleApp/CustomerCollection.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Collections.ObjectModel;
using WpfEfDAL;


namespace WpfEfSampleApp
{
    class CustomerCollection : MyBaseCollection<Customer>
    {
        /// <summary>
        // Return true if any customer in this collection has a validation error
        /// <summary>
        public Boolean HasErrors
        {
            get {
                return (from cust in this where cust.HasErrors select cust).Count() > 0;
            }
        }

        public CustomerCollection(IEnumerable<Customer> customers, OMSEntities context):base(customers, context)
        {
        }

        protected override void InsertItem(int index, Customer item)
        {
            if (! this.IsLoading)
            {
                // Tell the ObjectContext to start tracking this customer entity
                this.Context.AddToCustomers(item);
            }
            
            base.InsertItem(index, item);
        }

        protected override void RemoveItem(int index)
        {
            // Tell the ObjectContext to delete this customer entity
            this.Context.DeleteObject(this[index]);
            base.RemoveItem(index);
        }
    }
}

File: WPF Databinding in Visual Studio 2010 RTM/[VB]-WPF Databinding in Visual Studio 2010 RTM/VB/WpfEfSampleApp/CustomerCollection.vb
Imports System.Collections.ObjectModel
Imports WpfEfDAL

Public Class CustomerCollection
    Inherits MyBaseCollection(Of Customer)

    ''' <summary>
    ''' Return true if any customer in this collection has a validation error
    ''' </summary>
    Public ReadOnly Property HasErrors() As Boolean
        Get
            Return (Aggregate cust In Me Where cust.HasErrors Into Count()) > 0
        End Get
    End Property

    Sub New(ByVal customers As IEnumerable(Of Customer), ByVal context As OMSEntities)
        MyBase.New(customers, context)
    End Sub

    Protected Overrides Sub InsertItem(ByVal index As Integer, ByVal item As Customer)
        If Not Me.IsLoading Then
            'Tell the ObjectContext to start tracking this customer entity
            Me.Context.AddToCustomers(item)
        End If
        MyBase.InsertItem(index, item)
    End Sub

    Protected Overrides Sub RemoveItem(ByVal index As Integer)
        'Tell the ObjectContext to delete this customer entity
        Me.Context.DeleteObject(Me(index))
        MyBase.RemoveItem(index)
    End Sub

End Class
